Zend/Optimizer/zend_optimizer.c

```diff
@@ -831,6 +831,51 @@ zend_class_entry *zend_optimizer_get_class_entry_from_op1(
 	return NULL;
 }
 
+const zend_class_constant *zend_fetch_class_const_info(
+	const zend_script *script, const zend_op_array *op_array, const zend_op *opline, bool *is_prototype) {
+	const zend_class_entry *ce = NULL;
+	bool is_static_reference = false;
+
+	if (!opline || !op_array || !script || opline->op2_type != IS_CONST || Z_TYPE_P(CRT_CONSTANT(opline->op2)) != IS_STRING) {
```

Comment:
Checking for ``!script`` disables type inference for tracing JIT (including type-inference for ``self``).

Without script you may lookup in EG(class_table) and then check if the found class is defined in the same source file.

Reply:
Thanks for catching this! I've just attempted to improve the tracing JIT support